pkg/helpers/helpers.go

```diff
@@ -0,0 +1,641 @@
+// Copyright (c) Red Hat, Inc.
+// Copyright Contributors to the Open Cluster Management project
+
+package helpers
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"os"
+	"strconv"
+	"strings"
+	"text/template"
+
+	clusterv1 "github.com/open-cluster-management/api/cluster/v1"
+	operatorv1 "github.com/open-cluster-management/api/operator/v1"
+	workv1 "github.com/open-cluster-management/api/work/v1"
+	"github.com/open-cluster-management/managedcluster-import-controller/pkg/constants"
+
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	rbacv1 "k8s.io/api/rbac/v1"
+	crdv1 "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1"
+	crdv1beta1 "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1beta1"
+	"k8s.io/apimachinery/pkg/api/equality"
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/api/meta"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/runtime/serializer"
+	"k8s.io/apimachinery/pkg/types"
+	utilerrors "k8s.io/apimachinery/pkg/util/errors"
+	utilruntime "k8s.io/apimachinery/pkg/util/runtime"
+	"k8s.io/apimachinery/pkg/util/version"
+	"k8s.io/client-go/tools/clientcmd"
+	clientcmdapi "k8s.io/client-go/tools/clientcmd/api"
+	"k8s.io/client-go/tools/record"
+	"k8s.io/klog/v2"
+
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/client/apiutil"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+)
+
+const maxConcurrentReconcilesEnvVarName = "MAX_CONCURRENT_RECONCILES"
+
+var v1APIExtensionMinVersion = version.MustParseGeneric("v1.16.0")
+
+var crdGroupKind = schema.GroupKind{Group: "apiextensions.k8s.io", Kind: "CustomResourceDefinition"}
+
+var (
+	genericScheme = runtime.NewScheme()
+	genericCodecs = serializer.NewCodecFactory(genericScheme)
+	genericCodec  = genericCodecs.UniversalDeserializer()
+)
+
+func init() {
+	utilruntime.Must(appsv1.AddToScheme(genericScheme))
+	utilruntime.Must(corev1.AddToScheme(genericScheme))
+	utilruntime.Must(rbacv1.AddToScheme(genericScheme))
+	utilruntime.Must(crdv1beta1.AddToScheme(genericScheme))
+	utilruntime.Must(crdv1.AddToScheme(genericScheme))
+	utilruntime.Must(operatorv1.AddToScheme(genericScheme))
+}
+
+// GetMaxConcurrentReconciles get the max concurrent reconciles from MAX_CONCURRENT_RECONCILES env,
+// if the reconciles cannot be found, return 1
+func GetMaxConcurrentReconciles() int {
+	maxConcurrentReconciles := 1
+	if os.Getenv(maxConcurrentReconcilesEnvVarName) != "" {
+		var err error
+		maxConcurrentReconciles, err = strconv.Atoi(os.Getenv(maxConcurrentReconcilesEnvVarName))
+		if err != nil {
+			klog.Warningf("The value of %s env is wrong, using default reconciles (1)", maxConcurrentReconcilesEnvVarName)
+			maxConcurrentReconciles = 1
+		}
+	}
+	return maxConcurrentReconciles
+}
+
+// SetLabels set a label on the given object
+func SetLabels(objMeta *metav1.ObjectMeta, key, val string) bool {
```

Comment:
suggest to refer this method https://github.com/openshift/library-go/blob/master/pkg/operator/resource/resourcemerge/object_merger.go#L138  can update/add/remove labels.

Reply:
Yeah, after I imported the `library-go`, I will use it

Reply:
remove this two method, will use MergeMap directly